### ISSUE TITLE
Adding MakerNote parsing for Reconyx

### ISSUE THREAD

**Issue by parappathekappa** [ianare/exif-py#118]
Hey there,

Very clean repo!  Excited to contribute.

It looks like a Reconyx camera that we're using doesn't have the mapping setup, which isn't surprising.  I'm happy to do the legwork, but I wanted to get a point in the right direction first.  The mfg specific mapping seems hard-coded in the classes.py file, and so I didn't want to head down a wrong direction.

I was able to find the mfg specific tag indexes for the MakerNotes, so it seems like the hard work is done.  If you don't mind just giving me some suggestions on how to implement, I'd be happy to submit a PR, or fork and commit if that is the preferred method for contributing.

-Bryan

**Comment by ianare** [comment]
Thanks much for the offer, I'll be glad to merge a PR you can provide on this. If you can also provide test images with your PR that would be great.

Right now the code is a bit old and tangled... I've done some cleanup work and added a linter but more could be done.

In any case you're right, the manufacturer-specific definitions should go in their own file inside the `makernote` directory. The mapping / offset stuff is done in the `classes.py` file which I'm not really satisfied with but it does have the advantage of being relatively easy to understand.

**Comment by parappathekappa** [comment]
I was going to say how clean and easy to read the code is!

I've got the manufacturer-specific definitions created in the makernote
directory.  Unfortunately I have to add some logic in classes to try and
identify these reconyx files, since they don't include a make tag in the
exif data, which seems really strange.  They also don't include any
timestamp information in their exif tags either, which is why I'm having to
do all this in the first place.  I'll do a first pass but I'd love to get
some feedback on the logic or maybe an assist.

**Comment by ianare** [comment]
Yes that's pretty odd that they don't include the Maker info.... can you send me a couple sample files? I'd like to add them to the test repo.
https://github.com/ianare/exif-samples

**Comment by parappathekappa** [comment]
Sorry for the delay.  Here is one of the images.

[image: IMG_0028.JPG]

This site was actually able to parse the makernote data for this file:
metapicz.com. However, I just tried it again and it looks like that site is
down.

I'll try and find some time to push a PR in the next couple weeks.

-Bryan

**Comment by ianare** [comment]
No worries on the delay.

I think you need to use the Github site rather than email for sending the image though. I don't see it in the message in any case.

Alternatively, you can always create a PR on the samples repo. Up to you.

**Comment by parappathekappa** [comment]
Done!  I just submitted a PR.  I will also submit a PR for the exif-py repo
even though it's not a working solution yet.  This way you could see what
I've started on and see if you have any suggestions before I proceed.

-Bryan
